pinot-spi/src/main/java/org/apache/pinot/spi/stream/OffsetCriteria.java

```diff
@@ -168,24 +183,30 @@ public OffsetCriteria withOffsetCustom(String customString) {
      */
     public OffsetCriteria withOffsetString(String offsetString) {
```

Comment:
To be consistent with other types, could you add method `withOffsetAsTimestamp`?